src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
@cjcenizal If we did decide to go down the flux/redux path, would you change your mind?  It sounds like you wouldn't, that you just don't like this approach. Whichever direction we decide to take, we are making a decision, not avoiding making a decision.  We either decide to keep state and actions separate, or decide to group them together.

I think now *is* the time to have this conversation/debate and I think it's important to keep in mind possible decisions made regarding state.

I like both approaches, and see the merit in both, but I think we should figure out now which route to go down, and not rule one out without even considering it.  Maybe we can come up with a compromise solution?

I uploaded some changes:
  - Storing state inside actions to avoid the need to pass it around everywhere.
  - Created a LandingPageTable inside `ui/public`.  Passing it tableActions to avoid the slew of function parameters.  Building the table inside there rather than ui_framework.  This was an attempt to take your `ControlledTable` into account without duplicating code for dashboard/visualize.  I haven't yet, but if you like this solution, I would get rid of `visualize_landing_page_table.js` entirely.

I only made this changes going down the visualize path.  The dashboard path I don't think is working right now - it's still using the previous LandingPageTable inside ui_framework.

To see the changes, specifically look at:
 - `saved_object_table\landing_page_table.js`
- `saved_object_table\item_table_actions`
- and maybe `visualize\listing\visualize_listing.js`